apps/picsa-tools/monitoring-tool/src/app/pages/form/form-view/form-view.component.ts

```diff
@@ -24,184 +31,297 @@ export class FormViewComponent implements OnInit, OnDestroy {
     model: string;
     /** DB submission */
     submission: IFormSubmission;
-  };
+  } | null = null;
 
   /** Form entry data from enketo form */
   public formEntry?: IEnketoFormEntry;
-  private formId: string;
+
+  public isLoading = signal(true);
+  public isAuthenticating = signal(false);
+
+  private formId = '';
+  private currentForm: IMonitoringForm | null = null;
+
   /** Track if form has already had finalisation action (e.g. update/delete) */
   private formFinalised = false;
 
   private formInteracted = false;
 
   /** DB doc linked to current submission entry */
-  private submissionDoc: RxDocument<IFormSubmission>;
-  private componentDestroyed$ = new Subject<boolean>();
+  private submissionDoc?: RxDocument<IFormSubmission>;
+  private componentDestroyed$ = new Subject<void>();
 
   constructor(
     private route: ActivatedRoute,
+    private router: Router,
     private monitoringService: MonitoringToolService,
     private componentService: PicsaCommonComponentsService,
-    private dialogService: PicsaDialogService
+    private dialogService: PicsaDialogService,
+    private dialog: MatDialog,
+    private snackBar: MatSnackBar
   ) {}
 
-  async ngOnDestroy() {
+  // Track user activity to reset inactivity timer
+  @HostListener('document:mousemove')
+  @HostListener('document:keypress')
+  @HostListener('document:click')
+  @HostListener('document:touchstart')
+  onUserActivity(): void {
```

Comment:
nit(non-blocking): nice feature to add for the autoLock, although I think you probably don't need to listen for all of these interactions. I think likely can just use the `handleAutoSave` callback to reset whenever the form is saved locally